### PR TITLE
swarm - feat: add image generation interface and base implementation

### PR DESCRIPTION
**Pull request by MichaelDecent**

### CODE REVIEW

**Comment by MichaelDecent** [comment]
[](https://github.com/orgs/swarmauri/projects/13/views/1?pane=issue&itemId=80625354&issue=swarmauri%7Cswarmauri-sdk%7C465)

File: pkgs/swarmauri/swarmauri/image_gens/base/ImageGenBase.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from pydantic import ConfigDict, model_validator, Field
+from swarmauri_core.image_gens.IGenImage import IGenImage
+from swarmauri_core.ComponentBase import ComponentBase, ResourceTypes
+
+
+class ImageGenBase(IGenImage, ComponentBase):
+    allowed_models: List[str] = []
+    resource: Optional[str] = Field(default=ResourceTypes.IMAGE_GEN.value, frozen=True)
+    model_config = ConfigDict(extra="forbid", arbitrary_types_allowed=True)
+    type: Literal["LLMBase"] = "LLMBase"
```
Comment (cobycloud) [inline pkgs/swarmauri/swarmauri/image_gens/base/ImageGenBase.py]: type: Literal["ImageGenBase"] = "ImageGenBase"

File: pkgs/swarmauri/swarmauri/image_gens/concrete/BlackForestImgGenModel.py [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+
+    asyncio: ClassVar = asyncio
+    name: str = "flux-pro"  # Default model
+    type: Literal["BlackForestImgGenModel"] = "BlackForestImgGenModel"
+
+    def __init__(self, **data):
+        """
+        Initializes the BlackForestImgGenModel instance with HTTP clients.
+        """
+        super().__init__(**data)
```
Comment (cobycloud) [inline pkgs/swarmauri/swarmauri/image_gens/concrete/BlackForestImgGenModel.py]: change `**data` to `**kwargs`

move `super().__init__(**kwargs)` to the end of this method

**Review by cobycloud — CHANGES_REQUESTED** [review]
small changes requested, but great work in general

File: pkgs/swarmauri/swarmauri/image_gens/concrete/BlackForestImgGenModel.py (hunk 2 above)
Comment (MichaelDecent) [reply pkgs/swarmauri/swarmauri/image_gens/concrete/BlackForestImgGenModel.py]: Okay. Noted!. 
Also i will be moving all Image Gen Models, to the `image_gens/concrete/`

File: pkgs/swarmauri/swarmauri/image_gens/base/ImageGenBase.py (hunk 1 above)
Comment (cobycloud) [reply pkgs/swarmauri/swarmauri/image_gens/base/ImageGenBase.py]: please see my changes

**Review by cobycloud — APPROVED** [review]

**Review by cobycloud — APPROVED** [review]

**Pull request merged**